### PR TITLE
Prevent SELECT INSTR error in Postgres logs every minute (bsc#1157034)

### PR DESCRIPTION
**Pull request by moio**
## What does this PR change?

It prevents the following error from appearing in `/var/lib/pgsql/data/log/postgresql-$DATE-$TIME.log` every minute:

```
2019-11-25 15:11:08.665 CET susemanager spacewalk [13023]ERROR:  function rpm.rpmstrcmp(integer, unknown) does not exist at character 8
2019-11-25 15:11:08.665 CET susemanager spacewalk [13023]HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
2019-11-25 15:11:08.665 CET susemanager spacewalk [13023]STATEMENT:  select rpm.rpmstrcmp(1,'2');
2019-11-25 15:11:35.404 CET   [13673]ERROR:  function instr(character varying, unknown) does not exist at character 8
2019-11-25 15:11:35.404 CET   [13673]HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
2019-11-25 15:11:35.404 CET   [13673]QUERY:  SELECT INSTR(str1, '-') <> 0
2019-11-25 15:11:35.404 CET   [13673]CONTEXT:  PL/pgSQL function rpm.rpmstrcmp(character varying,character varying) line 30 at IF
        PL/pgSQL function rpm.vercmp(character varying,character varying,character varying,character varying,character varying,character varying) line 20 at assignment
        PL/pgSQL function public.evr_t_compare(public.evr_t,public.evr_t) line 3 at RETURN
        automatic analyze of table "susemanager.public.rhnpackageevr"
```

The error seems to stem from a PostgreSQL bug: when a function is executed in the context of the autoanalyze daemon (in this case a comparison over `evr_t`s, which are types we defined, it can't really use functions in the `public` namespace without explicitly prefixing them with `public.`.

In this case I preferred removing the need for the Oracle backwards-compatible `INSTR` function altogether by using the SQL standard `POSITION` ([docs](https://www.postgresql.org/docs/10/functions-string.html)).

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed: bugfix
- [x] **DONE**

## Test coverage
- No tests: already covered

- [x] **DONE**

## Links

Fixes https://bugzilla.suse.com/show_bug.cgi?id=1157034 partially
Tracks https://github.com/SUSE/spacewalk/pull/10148

- [x] **DONE**

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_lint_checkstyle"		 
- [ ] Re-run test "java_pgsql_tests" (Test skipped, there are no changes to test)		 
- [ ] Re-run test "ruby_rubocop"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"		 
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"		 
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Comment by moio** [comment]
@juliogonzalez can you just double check the upgrade directory is correct?

File: schema/spacewalk/susemanager-schema.changes [hunk 1]
```diff
@@ -1,4 +1,17 @@
...
+Thu Oct 31 15:12:19 CET 2019 - jgonzalez@suse.com
+
+- version 4.0.16-1
+- Fix: regression with Ubuntu version compare (bsc#1150113)
+
+-------------------------------------------------------------------
+Thu Oct 17 13:12:44 CEST 2019 - jgonzalez@suse.com
+
+- version 4.0.15-1
+>>>>>>> efc8f462f6f... Prevent SELECT INSTR error in Postgres logs every minute (bsc#1157034)
```
Comment (juliogonzalez) [inline schema/spacewalk/susemanager-schema.changes]: Conflict! Please add only your line on top of the file.

File: schema/spacewalk/upgrade/susemanager-schema-4.0.16-to-susemanager-schema-4.0.17/001-rpm.pkb.sql [hunk 2]
```diff
@@ -0,0 +1,249 @@
+-- create schema rpm;
```
Comment (juliogonzalez) [inline schema/spacewalk/upgrade/susemanager-schema-4.0.16-to-susemanager-schema-4.0.17/001-rpm.pkb.sql]: Incorrect folder for the script!

With this the users with Uyuni installed and already at version 4.1.x would not get this SQL executed.

Use `susemanager-schema-4.1.1-to-susemanager-schema-4.1.2` (you will need to create it as I just tagged).

File: schema/spacewalk/susemanager-schema.changes (hunk 1 above)
Comment (moio) [reply schema/spacewalk/susemanager-schema.changes]: Argh! fixed

File: schema/spacewalk/upgrade/susemanager-schema-4.0.16-to-susemanager-schema-4.0.17/001-rpm.pkb.sql (hunk 2 above)
Comment (moio) [reply schema/spacewalk/upgrade/susemanager-schema-4.0.16-to-susemanager-schema-4.0.17/001-rpm.pkb.sql]: Thanks. Fixed

**Pull request merged**